### PR TITLE
Improve `redis.utils` types

### PR DESCRIPTION
**Pull request by sobolevn**
Ref: https://github.com/redis/redis-py/blob/b2bcbc8bf1ae5622567acbc20e30e9c2e977e5c0/redis/utils.py
Unused function issue: https://github.com/redis/redis-py/issues/2299

### CODE REVIEW

File: stubs/redis/redis/utils.pyi [hunk 1]
```diff
@@ -14,11 +15,8 @@ def from_url(url: str, *, db: int = ..., decode_responses: Literal[True], **kwar
...
-@overload
-def str_if_bytes(value: bytes) -> str: ...  # type: ignore[misc]
-@overload
-def str_if_bytes(value: _T) -> _T: ...
+def str_if_bytes(value: str | bytes) -> str: ...
 def safe_str(value: object) -> str: ...
-def dict_merge(*dicts): ...
-def list_keys_to_dict(key_list, callback): ...
-def merge_result(command, res): ...
+def dict_merge(*dicts: Mapping[str, object]) -> dict[str, object]:
```
Comment (AlexWaygood) [inline stubs/redis/redis/utils.pyi:20]:
```suggestion
def dict_merge(*dicts: Mapping[str, object]) -> dict[str, object]: ...
```

File: stubs/redis/redis/utils.pyi [hunk 2]
```diff
@@ -14,11 +15,8 @@ def from_url(url: str, *, db: int = ..., decode_responses: Literal[True], **kwar
...
-@overload
-def str_if_bytes(value: bytes) -> str: ...  # type: ignore[misc]
-@overload
-def str_if_bytes(value: _T) -> _T: ...
+def str_if_bytes(value: str | bytes) -> str: ...
 def safe_str(value: object) -> str: ...
-def dict_merge(*dicts): ...
-def list_keys_to_dict(key_list, callback): ...
-def merge_result(command, res): ...
+def dict_merge(*dicts: Mapping[str, object]) -> dict[str, object]: ...
```
Comment (AlexWaygood) [inline stubs/redis/redis/utils.pyi:20]: We can probably be a bit more sophisticated here — if all the input `Mapping`s have `int` as their value type, the returned `dict` will also have `int` as its value type:

```suggestion
def dict_merge(*dicts: Mapping[str, _T]) -> dict[str, _T]: ...
```

Comment (AlexWaygood) [reply stubs/redis/redis/utils.pyi:20]: But then `_T` just gets solved as `object`, which is no worse than what you have now

**Review by AlexWaygood — APPROVED** [review]

**Comment by sobolevn** [comment]
Thank you! 👍

**Pull request merged**